src/lib/auth/accountKeys.mjs

```diff
@@ -0,0 +1,130 @@
+import { container } from "../../cli.mjs";
+import { FaunaAccountClient } from "../fauna-account-client.mjs";
+import { AccountKeyStorage } from "../file-util.mjs";
+import { InvalidCredsError } from "../misc.mjs";
+
+/**
+ * Class representing the account key(s) available to the user.
+ * This class is scoped to a specific profile, as each command invocation will correlate
+ * 1:1 with a profile. The profile determines how we access the local credentials file.
+ *
+ * Keeps track of local keys in this.keyStore, used for getting and saving to/from the filesystem.
+ * this.key is the currently active account key, it stays updated after refreshes
+ */
+export class AccountKeys {
+  constructor(argv) {
+    this.logger = container.resolve("logger");
+    this.profile = argv.profile;
+    this.keyStore = new AccountKeyStorage(this.profile);
+    const storedKey = this.keyStore.get()?.accountKey;
+    const { key, keySource } = AccountKeys.resolveKeySources(argv, storedKey);
+    this.key = key;
+    this.keySource = keySource;
+
+    // Let users know if the creds they provided are invalid (empty)
+    if (!key && keySource !== "credentials-file") {
+      throw new Error(
+        `The account key provided by ${keySource} is invalid. Please provide an updated value.`,
+      );
+    }
+  }
+
+  /**
+   * Evaluates the accountKey to use for this instance, based on priority order
+   * @param {Object} argv yargs arguments
+   * @param {string | undefined} storedKey The account key stored in the credentials file
+   * @returns {Object} - The account key and its source
+   */
+  static resolveKeySources(argv, storedKey) {
+    let key, keySource;
+    // argv.accountKey can come from flag, config, or FAUNA_ACCOUNT_KEY
+    if (argv.accountKey) {
+      key = argv.accountKey;
+      keySource = "user";
+    } else {
+      key = storedKey;
+      keySource = "credentials-file";
+    }
+    return {
+      key,
+      keySource,
+    };
+  }
+
+  /**
+   * Prompt re-authentication and exit the program;
+   */
+  promptLogin() {
+    const exit = container.resolve("exit");
+    this.logger.stderr(
+      `The requested profile ${this.profile || ""} is not signed in or has expired.\nPlease re-authenticate`,
+    );
+    this.logger.stdout(`To sign in, run:\n\nfauna login\n`);
+    exit(1);
+  }
```

Comment:
Is this the blessed why for exiting when you want to control the messaging?

Asking for a friend. 🤭

Reply:
You can very well just `throw new Error(message)` to exit and control the messaging. But it goes to `stderr` only. I guess I just wanted to have a blip of `stdout` after? 

Should probably just throw

Reply:
we want to avoid calling `exit` in random places